gobblin-core/src/main/java/gobblin/publisher/JdbcPublisher.java

```diff
@@ -199,5 +179,28 @@ public void publishData(Collection<? extends WorkUnitState> states) throws IOExc
   }
 
   @Override
-  public void publishMetadata(Collection<? extends WorkUnitState> states) throws IOException {}
+  public void publishMetadata(Collection<? extends WorkUnitState> states) throws IOException {
+    // Nothing to do
+  }
+  
+  @Override
+  public void publishMetadata(WorkUnitState state) throws IOException {
+    // Nothing to do
+  }
+  
+  @Override
+  public void publishData(WorkUnitState state) throws IOException {
```

Comment:
This method is only called from the tasks. If I understand correctly there are two options:

1. JDBC writer wrote directly to main table. This (for now) is equivalent to `commitPolicy!=COMMIT_ON_FULL_SUCCESS`. In this case, the message should probably simply be something like `Records written directly to table. No publish necessary.`. However, there's the problem of guaranteeing that that was the case, which right now relies on a single key, and a developer may mistakenly change that in the future in the writer but not in the publisher. Instead, I see two options:
1.1. Have a static method in the writer that determines whether it writes directly to main table. Use that method from both writer and publisher.
1.2. (Preferable) Write whether the data is already published directly on the `WorkUnitState`. An existing way of doing this is using the `FinalState` functionality (see for example `FsDataWriter`, specifically the `getFinalState()` method.

2. JDBC writer used a staging table. Currently, this guarantees the commit to be done at the job level, and this method should never be called. However, this may not be the case in the future (again, developer might unknowingly modify the writer). If this method should not have been called, then the method should throw an error and fail the task. If it should have been called, then it should do the appropriate publishing work.